Citavi 5 XML.js

```diff
@@ -0,0 +1,776 @@
+		
+		var publishers = ZU.xpathText(doc, '//ReferencePublishers/OnetoN[starts-with(text(), "'+item.itemID+'")]');
+		if (publishers && publishers.length>0) {
+			item.publisher = attachName(doc, publishers).join('; ');
+		}
+		
+		var keywords = ZU.xpathText(doc, '//ReferenceKeywords/OnetoN[starts-with(text(), "'+item.itemID+'")]');
+		if (keywords && keywords.length>0) {
+			item.tags = attachName(doc, keywords);
+		}
+		
+		var citations = ZU.xpath(doc, '//KnowledgeItem[ReferenceID="'+item.itemID+'"]');
+		for (var j=0; j<citations.length; j++) {
+			var title = ZU.xpathText(citations[j], 'CoreStatement');
+			var text = ZU.xpathText(citations[j], 'Text');
+			if (title) {
+				text = '<h1>' + title + '</h1>' + text;
+			}
+			if (text) {
+				item.notes.push({note : text});
+			}
+		}
+		
+		itemIdList[item.itemID] = item;
+		if (type == "Contribution") {
+			unfinishedReferences.push(item);
+		} else {
+			item.complete();
+		}
+	}
+	
+	//
+	for (var i=0; i< unfinishedReferences.length; i++) {
+		var item = unfinishedReferences[i];
+		var containerString = ZU.xpathText(doc, '//ReferenceReferences/OnetoN[contains(text(), "'+item.itemID+'")]');
+		if (containerString) {
+			var containerId = containerString.split(';')[0];
+			var containerItem = itemIdList[containerId];
+			if (containerItem.type == "ConferenceProceedings") {
+				item.itemType = "conferencePaper";
+			}
+			item.publicationTitle = containerItem.title;
+			item.place = containerItem.place;
+			item.puslisher = containerItem.publisher;
+			item.ISBN = containerItem.ISBN;
+			//TODO editors and maybe more
+			
+		}
+		item.complete();
+	}
+	
+	var categories = ZU.xpath(doc, '//Categories/Category');
+	var collectionList = [];
+	for (var i=0, n=categories.length; i<n; i++) {
+		var collection = new Zotero.Collection();
+		collection.name = ZU.xpathText(categories[i], './Name');Z.debug(collection.name);
+		collection.id = ZU.xpathText(categories[i], './@id');
+		collection.type = 'collection';
+		collection.children = [];
+		var referenceCategories = ZU.xpath(doc, '//ReferenceCategories/OnetoN[contains(text(), "'+collection.id+'")]');
+		for (var j=0; j<referenceCategories.length; j++) {
+			var refid = referenceCategories[j].textContent.split(';')[0];
+			collection.children.push({ type: 'item', id: refid });
+		}
+		collectionList.push(collection);
+		collection.complete();//Change if subcollection work...
+	}
+	/* TODO fix this, but how?
+	var openCollections = [];
+	for (var i=0; i<collectionList.length; i++) {
+		var collection = collectionList[i];
+		var categoryCategories = ZU.xpathText(doc, '//CategoryCatgories/OnetoN[starts-with(text(), "'+collection.id+'")]');
+		if (categoryCategories) {
+			var subcategories = categoryCategories.split(';');
+			Z.debug(collection.id);
+			Z.debug(subcategories);
+			for (var j=1; j<subcategories.length; j++) {
+				collection.children.push({ type: 'collection', id: subcategories[j] });
```

Comment:
My impression is that everything here works, but the `complete()`-statements are not clear for me...